core/blockchain.go

```diff
@@ -276,12 +276,12 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, chainConfig *par
 
 	if bc.empty() {
 		rawdb.InitDatabaseFromFreezer(bc.db)
-		// If ancient database is not empty, reconstruct all missing
-		// indices in the background.
-		frozen, _ := bc.db.Ancients()
-		if frozen > 0 {
-			txIndexBlock = frozen
-		}
+	}
+	// If ancient database is not empty, reconstruct all missing
+	// indices in the background.
+	frozen, _ := bc.db.Ancients()
+	if frozen > 0 {
+		txIndexBlock = frozen
```

Comment:
It's not correct, previously we will only re-init the ancient blocks in this special corner case where state is empty but ancient store it not. With this change we will re-init the ancient blocks all the time.

I will try to figure out a proper solution to fix it.